Add App configuration AFD template (#14668)
Azure/azure-quickstart-templates: quickstarts/microsoft.appconfiguration/app-configuration-afd/README.md

---

> description: This template creates an Azure FrontDoor resource, an Azure Front Door endpoint with the existing App Configuration store set as the origin, sets ups the route and rules that control which requests pass through Azure Front Door to App Configuration. The template also connects the App Configuration store to the newly created Azure Front Door resource.

Reviewer: ```suggestion
description: The template connects Azure App Configuration store to a newly created Azure Front Door resource. The template creates a Front Door profile, a Front Door endpoint with App Configuration as origin, and sets up a route with rules that control which requests pass through Azure Front Door to App Configuration. 
```

![Cred Scan Check](https://azurequickstartsservice.blob.core.windows.net/badges/quickstarts/microsoft.appconfiguration/app-configuration-afd/CredScanResult.svg)

![Bicep Version](https://azurequickstartsservice.blob.core.windows.net/badges/quickstarts/microsoft.appconfiguration/app-configuration-afd/BicepVersion.svg)

[![Deploy To Azure](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/deploytoazure.svg?sanitize=true)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2Fquickstarts%2Fmicrosoft.appconfiguration%2Fapp-configuration-afd%2Fazuredeploy.json)  [![Visualize](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/visualizebutton.svg?sanitize=true)](http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2Fquickstarts%2Fmicrosoft.appconfiguration%2Fapp-configuration-afd%2Fazuredeploy.json)

> This template creates an Azure Front Door resource and connects it to an existing App Configuration store. It also configures Front Door child resources and assigns the App Configuration Data Reader role to the Front Door managed identity.

Reviewer: If a new store can be created through the template, this line needs to be updated

> **Note:** This template does not create an App Configuration store or modify its key-values.

Author: Yes that is right

> **Note:** This template does not create an App Configuration store or modify its key-values.

Author: Updated

[![Deploy To Azure](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/deploytoazure.svg?sanitize=true)](https://portal.azure.com/#create/Microsoft.Template/uri/https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2Fquickstarts%2Fmicrosoft.appconfiguration%2Fapp-configuration-afd%2Fazuredeploy.json)  [![Visualize](https://raw.githubusercontent.com/Azure/azure-quickstart-templates/master/1-CONTRIBUTION-GUIDE/images/visualizebutton.svg?sanitize=true)](http://armviz.io/#/?load=https%3A%2F%2Fraw.githubusercontent.com%2FAzure%2Fazure-quickstart-templates%2Fmaster%2Fquickstarts%2Fmicrosoft.appconfiguration%2Fapp-configuration-afd%2Fazuredeploy.json)

This template creates an Azure App Configuration store if it doesn't exist, an Azure Front Door resource and connects it to the App Configuration store. It also configures Front Door child resources and assigns the App Configuration Data Reader role to the Front Door managed identity.

> **Note:** This template creates an App Configuration store if the provided store name does not exist.

Reviewer: Sure
